src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -122,35 +127,36 @@ public void Allocate(int capacity)
         /// <returns>Cloned packet</returns>
         public SNIPacket Clone()
         {
-            SNIPacket packet = new SNIPacket(null);
-            packet._data = new byte[_length];
-            Buffer.BlockCopy(_data, 0, packet._data, 0, _length);
-            packet._length = _length;
-
-            return packet;
+            SNIPacket clonedPacket = new SNIPacket();
+            clonedPacket._dataBuffer = new byte[_dataBuffer.Length];
```

Comment:
There is a behavior change. The _length was pointing to the length of the available amount of data. 
However the change allocates a new byte array with the complete length of the old array. Was this a bug earlier or can this introduce a new bug now? Or does it not matter?